conans/test/utils/tools.py

```diff
@@ -482,16 +482,17 @@ def __init__(self, base_folder=None, current_folder=None,
 
         if cpu_count:
             self.client_cache.conan_config
-            replace_in_file(os.path.join(self.client_cache.conan_conf_path),
+            replace_in_file(self.client_cache.conan_conf_path,
                             "# cpu_count = 1", "cpu_count = %s" % cpu_count,
-                            output=TestBufferConanOutput())
+                            output=TestBufferConanOutput(), strict=False)
```

Comment:
why strict=False? This replacement should happen, is there any reason it does not?
Putting ``strict=False`` could silently skip a change we do on the conan.conf default file, and then suddenly our test suite starts to randomly fail again.

Reply:
If the check for the `cpu_count` in the `conan.conf` is based on the substitution performed inside the `TestClient` class... that is not a test. BTW, the same applies to the `revisions_enabled` conf variable.

I need this change to be able to create two TestClient objects using the same cache. Here are some tests using one client for the package in editable mode and another for a consuming project, I think this is clearer than having only one client and jumping between two different folders to call commands.

Reply:
Maybe use ``strict=base_folder``? In this way, if it is a new TestClient will replace or fail, and if it is another TestClient that reuses the cache in ``base_folder`` will not fail.

Reply:
Ok, I can do a `strict = not bool(base_folder)` then (here and for revisions too).